frontend/src/lib/get-cluster.ts

```diff
@@ -386,7 +399,14 @@ export function getClusterStatus(
                 activeCsr && !activeCsr?.status?.certificate ? ClusterStatus.needsapproval : ClusterStatus.pendingimport
         }
     } else {
-        mcStatus = clusterAvailable ? ClusterStatus.ready : ClusterStatus.offline
+        if (clusterAvailable) {
+            const hasDegradedAddons = !!managedClusterAddOns?.some((mca) =>
+                checkForCondition(AddonStatus.Degraded, mca.status?.conditions!)
```

Comment:
will this check all addon status and return true if any one of them has Degraded ?

Reply:
Yeah the `.some` in JS will return true if anything in the array returns true based on the condition